osu.Game/Scoring/ScoreManager.cs

```diff
@@ -163,11 +197,25 @@ public override bool IsAvailableLocally(ScoreInfo model)
 
         public Task<IEnumerable<Live<ScoreInfo>>> Import(ProgressNotification notification, ImportTask[] tasks, ImportParameters parameters = default) => scoreImporter.Import(notification, tasks);
 
-        public Task Export(ScoreInfo score) => scoreExporter.ExportAsync(score.ToLive(Realm));
+        /// <summary>
+        /// Export a replay from a given <see cref="IScoreInfo"/>.
+        /// </summary>
+        /// <param name="scoreInfo">The <see cref="IScoreInfo"/> to export.</param>
+        /// <returns>The <see cref="Task"/>. Null if the score on the database cannot be found.</returns>
+        /// <remarks>
+        /// The <see cref="IScoreInfo"/> is re-retrieved from the database to ensure all the required data
+        /// for exporting a replay are present (may have missing properties if it was retrieved from online data).
+        /// </remarks>
+        public Task? Export(ScoreInfo scoreInfo)
+        {
+            ScoreInfo? databasedScoreInfo = getDatabasedScoreInfo(scoreInfo);
+
+            return databasedScoreInfo == null ? null : scoreExporter.ExportAsync(databasedScoreInfo.ToLive(Realm));
+        }
```

Comment:
`Task?` is a very awkward return type to work with (you can't `await` it directly without nullchecking first etc). The established practice is to either return `Task.CompletedTask` or just fail the task by throwing.

I don't particularly see any value in handling imports where the databased score cannot be found so I'd be probably okay with silently not exporting anything but something should probably be logged in such case at least. Maybe even at important level to let the user know their score didn't export. Throwing an exception may be fine too.